src/rdkafka.h

```diff
@@ -3088,6 +3088,24 @@ rd_kafka_offsets_for_times (rd_kafka_t *rk,
                             int timeout_ms);
 
 
+/**
+ * @brief Allocate memory using the librdkafka internal allocator
+ *
+ * This is typically an abstraction for the malloc(3) call and makes sure
+ * the application can use the same memory allocator as librdkafka for
+ * allocating pointers that are used by librdkafka.
+ *
+ * \p rk must be set for memory returned by APIs that take an \c rk argument,
```

Comment:
This does not apply to malloc, only free.

Reply:
If we have the Handle* argument there we're free to start making use of it in the future, without asking users to fix their code.

Reply:
.. which gives us more options to work with in the future.